### ISSUE TITLE
How to use with a public repo?

### ISSUE THREAD

**Issue by skrobinson** [JacobDomagala/StaticAnalysis#28]
I would like to add StaticAnalysis as an Action to run on PR in a public repository (see https://github.com/p-ranav/argparse/pull/148#issuecomment-967472187). Is this possible?

The PR log seems to say that the action does not have permission to write to the repo to add a comment.

**Comment by JacobDomagala** [comment]
Hmm, I haven't seen this type of error before (I've been only using it on my own repositories). For now you can run the action with `force_console_print` input set to true, but it won't create the comments (it will only print the result to GitHub's action console log, see https://github.com/JacobDomagala/StaticAnalysis#non-pull-request)

**Comment by JacobDomagala** [comment]
Yes, I will have to fix that eventually, I wasn't aware of that issue before.

**Comment by JacobDomagala** [comment]
Ok, I was able to fix the issue with comments from forks (https://github.com/JacobDomagala/StaticAnalysis/pull/30), but there's another issue that I don't think I'll be able to resolve. 

The code snippets (or perma-links) work only on the same repository, meaning when you create PR from fork, and there're any issues found, the permalink to said file(s) will appear as link, not as a code snippet ([source](https://docs.github.com/en/github/writing-on-github/working-with-advanced-formatting/creating-a-permanent-link-to-a-code-snippet))

> This type of permanent link will render as a code snippet only in the repository it originated in. In other repositories, the permalink code snippet will render as a URL.

I'll try to see whether it's something that can be somehow worked around.

**Comment by skrobinson** [comment]
Are you seeing bot comment boxes with snippet URLs and the associated warning (or note)? Does clicking the URL open the PR head at the file with highlighted lines? If so, that's a good start and may be all that is possible with the current GH Actions API.

There is a [3+ year issue](https://github.community/t/embedding-a-code-snippet-broken/712) asking for this to be resolved. If GH does include a cross-repo code snippet feature in the future, SA is in a good position to use it.

Thank you for your work adding this feature.

P.S. I just noticed that code snippets do seem to work with review comments, that code should also originate in the PR head. Maybe this is an angle to investigate?

**Comment by JacobDomagala** [comment]
> Are you seeing bot comment boxes with snippet URLs and the associated warning (or note)? Does clicking the URL open the PR head at the file with highlighted lines?

Yes. [This](https://github.com/JacobDomagala/TestRepo/pull/7) is a test PR created from forked repo, that I'm using to test my changes. 

> I just noticed that code snippets do seem to work with review comments, that code should also originate in the PR head. Maybe this is an angle to investigate?

Initially I was thinking about generating the code block (using the Markdown) with the hyperlink to the source file. Something like :

-----------------------
**Issue found in** [source.cpp](https://github.com/JacobDTest/TestRepo/blob/d8ab771279aaad11f73b185d39159578a0f144e7/source.cpp#L1-L4)
```cpp
int main(int /*argc*/, char** argv){
    int unused = 0;
    return 0;
}
```
```diff
!Line: 2 - style: Variable 'unused' is assigned a value that is never used. [unreadVariable]
``` 
-----------------------
 

 But your idea could be easier to implement, I'll see if it works. Thanks!

**Comment by JacobDomagala** [comment]
Update: 
I ended up actually using my approach (creating code snippet using Markdown code-block), as it's easier to implement. The PR is close to being done, I need to double check the changes and update the documentation, so this should be done around weekend.

**Comment by skrobinson** [comment]
I'm not a fan of "<---- HERE", but it's not a problem, just a preference. Have you considered using a `diff` (rather than `cpp`) block with `!line` or `@@line@@`?

Example:

```diff
!   int anotherUnused;
}

int main(int /*argc*/, char** argv){
    int unused = 0;
```

Also, I looked over your recent commits on `make-it-work-on-fork` and my other questions/comments went away as I delved deeper. This new feature is coming together nicely. Thank you.
